postal/internal/dependency_mappings.go

```diff
@@ -1,52 +1,38 @@
 package internal
 
 import (
-	"fmt"
-	"os"
-	"path/filepath"
-	"strings"
+	"github.com/pkg/errors"
+
+	"github.com/paketo-buildpacks/packit/servicebindings"
 )
 
-type DependencyMappingResolver struct{}
+//go:generate faux --interface BindingResolver --output fakes/binding_resolver.go
+type BindingResolver interface {
+	Resolve(typ string, provider string) ([]servicebindings.Binding, error)
+}
 
-func NewDependencyMappingResolver() DependencyMappingResolver {
-	return DependencyMappingResolver{}
+type DependencyMappingResolver struct {
+	bindingResolver BindingResolver
 }
 
-// Reference file structure for bindings directory
-// - bindings
-//    - some-binding
-//       - type -> dependency-mapping
-// 			 - some-sha -> some-uri
-//       - other-sha -> other-uri
+func NewDependencyMappingResolver(bindingResolver BindingResolver) DependencyMappingResolver {
+	return DependencyMappingResolver{
+		bindingResolver: bindingResolver,
+	}
+}
 
-// Given a target dependency, look up if there is a matching dependency mapping at the given binding path
-func (d DependencyMappingResolver) FindDependencyMapping(sha256, bindingPath string) (string, error) {
-	allBindings, err := filepath.Glob(filepath.Join(bindingPath, "*"))
+// FindDependencyMapping looks up if there is a matching dependency mapping at the given binding path
+func (d DependencyMappingResolver) FindDependencyMapping(sha256 string) (string, error) {
+	bindings, err := d.bindingResolver.Resolve("dependency-mapping", "")
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "getting dependency-mapping binding")
```

Comment:
Rather than pulling in a whole new package, maybe we can use the stdlib built-in error wrapping?

```suggestion
		return "", fmt.Errorf("failed to resolve dependency-mapping binding: %w", err)
```

Reply:
Done here and elsewhere ✔️